lib/pocolog/stream_aligner.rb

```diff
@@ -99,6 +99,9 @@ def build_index(streams)
 	    Pocolog.info("got #{streams.size} streams with #{size} samples")
             tic = Time.now
             @base_time = streams.map { |s| s.stream_index.base_time }.min
+            if @base_time.nil?
```

Comment:
The only situations causing `@base_time` to be nil here are:
 1. 'streams' is empty
 2. all streams in 'streams' are empty
 3. a bug that cause a StreamIndex object to have a nil base_time even though it does have samples

Moreover, it looks like that having a single empty stream in the middle of a set of non-empty streams would cause an exception (since #base_time would be nil and nil cannot be compared with a number or time).

So, the proper fixes would be:
 - ignore all empty streams to compute time_intervals and base_time
 - raise if all streams are empty
 - if you have cases where a non-empty stream has an empty base time, verify for that at the point where the single stream was loaded, not here, where it's already way too late.